loadbalancing.md

```diff
@@ -0,0 +1,339 @@
+
+A loadbalancer is needed to route connections from your application(s) to a TiDB instance. This helps to distribute the load over multiple machines and also makes sure your application will be able to connect to another TiDB instance if an instance is unavailable for any reason.
+
+## Loadbalancer types
+
+The first type of loadbalancer that is available is *connector based loadbalancing*. Many of the MySQL connectors like MySQL Connector/J and MySQL Connector/C++. The benefits from this is that there is that there is no extra network hop and the application has more information about which server it is connected to. The drawback of this is that there is no central administration and when changing the configuration you have to do this on all your application hosts. Depending on the programming language you are using the loadbalancing may not be available or not offer more advanced options. This can work with most third-party applications as this is often configured in the connection string (JDBC url for Java for example).
+
+It is also possible to use DNS round-robin. However this is not advisable as this won't prevent connections from going to a machine that is unavailable. This means that your application may have to re-try connections more often if not all servers are available.
+
+Some of the newer MySQL connectors have support for [DNS SRV](https://dev.mysql.com/doc/refman/8.0/en/connecting-using-dns-srv.html). This is somewhat similar to round-robin DNS, but allows you to set priorities. The benefit from this is that it is an industry standard and that this makes the client-side configuration easier.
+
+If you are using Kubernetes you probably want to use the [LoadBalancer](https://docs.pingcap.com/tidb-in-kubernetes/stable/access-tidb#loadbalancer) service type that works with Amazon ELB and similar services from other cloud vendors. For on-premise Kubernetes something like [metallb](https://metallb.universe.tf/) can be used.
+
+When your are not using Kubernetes to deploy on a cloud service you can still use the loadbalancing service from your cloud provider.
+
+Another common option is to use a software based loadbalancer. Common ones are ProxySQL, haproxy and MySQL Router.
+
+The last type of loadbalancer that we will discuss here is hardware based loadbalancers. These are physical machines where you plug in a network cable. These are often costly, but also offer a high throughput.
+
+## General loadbalancer requirements for TiDB
+
+Where loadbalancers may offer advanced features like read/write splitting for MySQL, this is not needed for TiDB. The loadbalancer does not need to be able to inspect the MySQL protocol. A TCP level loadbalacer works fine for TiDB.
+
+The loadbalancer only needs access to the TiDB servers (TCP port 4000 by default) and doesn't need access to PD or TiKV servers.
+
+In general a TCP health check is sufficient. Health checks that check for `read_only` or other MySQL or InnoDB variables should be avoided.
```

Comment:
I think a HTTP health check is better along with configuration of `GracefulWaitBeforeShutdown`: https://github.com/pingcap/tidb/pull/20649

This helps ensure that topology changes have less impact on users sessions.

Reply:
I couldn't find this option in the docs, so I opened https://github.com/pingcap/docs/issues/5823

Reply:
You can also get into the same trouble if it is a duration, since an integer is valid and will be interpreted in nanoseconds :-)